src/snowflake/snowpark/session.py

```diff
@@ -265,6 +265,9 @@ def __enter__(self):
     def __exit__(self, exc_type, exc_val, exc_tb):
         self.close()
 
+    def __repr__(self):
```

Comment:
I will prefer to implement `__str__` instead of `__repr__`. If `__repr__` gets implemented, we have to make sure `eval(repr(x)) == x`. https://stackoverflow.com/questions/1436703/what-is-the-difference-between-str-and-repr